cpp/src/cylon/net/serialize.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CYLON_NET_TABLE_SERIALIZE_HPP
+#define CYLON_NET_TABLE_SERIALIZE_HPP
+
+#include <vector>
+
+namespace cylon {
+
+/**
+ * Serialize a table to send over the wire
+ */
+class TableSerializer {
+public:
+
+    virtual ~TableSerializer() = default;
+
+    /**
+     * get the buffer sizes for this table in bytes
+     * starting from column 0 to the last column
+     * For each column, three buffer sizes are returned in this order:
+     *      size of the column data buffer in bytes
+     *      size of the column null mask buffer in bytes
+     *      size of the column offsets buffer in bytes
+     * If there are two columns in a table, 6 data buffers sizes are returned
+     *      buffer[0] = size of the data buffer of the first column
+     *      buffer[1] = size of the null mask buffer of the first column
+     *      buffer[2] = size of the offsets buffer of the first column
+     *      buffer[3] = size of the data buffer of the second column
+     *      buffer[4] = size of the null mask buffer of the second column
+     *      buffer[5] = size of the offsets buffer of the second column
+     *
+     * If there are n columns, 3 * n buffer sizes are returned
+     *
+     * This method is symmetrical to getDataBuffers()
+     * @return
+     */
+    virtual std::vector<int32_t> getBufferSizes() = 0;
```

Comment:
Let's ask the `TableSerializer` to hold this buffers vector and just return a const ref of it to the outside. 
```suggestion
    virtual const std::vector<int32_t>& getBufferSizes() = 0;
```